server/src/main/java/io/spine/server/entity/DefaultRecordBasedRepository.java

```diff
@@ -43,19 +45,20 @@
      */
     protected DefaultRecordBasedRepository() {
         super();
-        EntityClass<E> entityClass = entityModelClass();
-        this.entityFactory = entityClass.factory();
-        TypeUrl stateType = entityClass.stateType();
-        this.storageConverter = DefaultConverter.forAllFields(stateType, this.entityFactory);
     }
 
     @Override
     protected EntityFactory<E> entityFactory() {
-        return this.entityFactory;
+        return entityModelClass().factory();
     }
 
     @Override
     protected StorageConverter<I, E, S> entityConverter() {
-        return this.storageConverter;
+        if (storageConverter == null) {
+            EntityClass<E> entityClass = entityModelClass();
+            TypeUrl stateType = entityClass.stateType();
+            storageConverter = DefaultConverter.forAllFields(stateType, entityFactory());
```

Comment:
Let's perform the initialization in `onRegistered()` method.

The only thing that the getter should do is to make sure the converter has been initialized when the method is called.